compact/tracking/tof_endcap.xml

```diff
@@ -133,48 +197,92 @@
+          <position x="EndcapTOF_Sensor_position" y="0" z="0" />
+        </module_component>
+        <module_component name="hybridkaptonbottom" material="Kapton" sensitive="false" width="EndcapTOF_Module_width" length="EndcapTOF_Module_length" thickness="EndcapTOF_Hybrid_thickness" vis="TOFHybridVis" >
+          <position x="EndcapTOF_Service_position" y="0" z="0" />
+        </module_component>
+        <module_component name="hybridtraces" material="Aluminum" sensitive="false" width="EndcapTOF_Module_width" length="EndcapTOF_Module_length" thickness="EndcapTOF_HybridTrace_thickness" vis="TOFCFoamVis" >
+          <position x="EndcapTOF_Service_position" y="0" z="0" />
+        </module_component>
+        <module_component name="hybridPCB" material="Fr4" sensitive="false" width="EndcapTOF_Module_width" length="EndcapTOF_Module_length" thickness="EndcapTOF_HybridPCB_thickness" vis="AnlGreen" >
+          <position x="EndcapTOF_Service_position" y="0" z="0" />
+        </module_component>
+        <module_component name="hybridkaptontop" material="Kapton" sensitive="false" width="EndcapTOF_Module_width" length="EndcapTOF_Module_length" thickness="EndcapTOF_Hybrid_thickness" vis="TOFHybridVis" >
+          <position x="EndcapTOF_Service_position" y="0" z="0" />
+        </module_component>
+      </moduleBack>
+      <layer module="EndcapTOF_Module1" id="1">
+        <envelope vis="TOF_envelope"
+          rmin="ForwardTOFLayer_rmin"
+          rmax="ForwardTOFLayer_rmax"
+          length="3*cm"
+          zstart="ForwardTOF_zmin" />
+        <layer_material surface="inner" binning="binPhi,binR" bins0="256" bins1="256"/>
+        <y_layout dr="0.0*mm" z0="EndcapTOF_zOffset" nz="EndcapTOF_Module_nz"/>
+        <z_layout z0="ForwardTOF_zmin"/>
+      </layer>
+      <support module="EndcapTOF_SupportDummy" id="1" vis="InvisibleWithDaughters">
+        <envelope vis="AnlGray"
+          rmin="ForwardTOFLayer_rmin"
+          rmax="ForwardTOFLayer_rmax"
+          length="12*mm"
+          zstart="ForwardTOF_zmin" />
+      </support>
+      <modsize
+          width="EndcapTOF_Module_width"
+          length="EndcapTOF_Module_length"
+          overlap="EndcapTOF_Module_overlap"
+          spacing="EndcapTOF_Module_spacing" />
+    </detector>
   </detectors>
 
   <plugins>
     <plugin name="DD4hep_ParametersPlugin">
       <argument value="ForwardTOF"/>
-      <argument value="layer_pattern: str=layer\d"/>
+      <argument value="layer_pattern: str=ForwardTOF_layer\d"/>
     </plugin>
   </plugins>
 
   <readouts>
     <readout name="TOFEndcapHits">
-      <segmentation type="CartesianGridXY" grid_size_x="100*um" grid_size_y="100*um" />
-      <id>system:8,layer:2,module:6,sensor:16,x:32:-16,y:-16</id>
+      <segmentation type="CartesianGridXY" grid_size_x="0.1*mm" grid_size_y="0.1*cm" />
+      <id>system:8,layer:4,module:8,idx:7,idy:5,x:32:-16,y:-16</id>
```

Comment:
Is there a corresponding PR to EICrecon that should be merged simultaneously? Changes in IDs may otherwise cause incompatibility.

Reply:
There is no problem with EICrecon. IDs are not used in reconstruction at the moment.